Angular/Angular guidelines and best practices/Formatting, naming and best practices.md

```diff
@@ -603,6 +603,45 @@ You can also write your own operators. If you do so, it is strongly recommended
 
 Covering all the various operators is out of the scope of this handbook. Please refer to some of the learning resources mentioned earlier [[1](/books/frontend/angular/getting-started-with-angular/get-to-know-rxjs)] [[2](/books/frontend/angular/angular-guidelines-and-best-practices/core-libraries-configuration-and-tools#a-hrefhttpsgithubcomreactivexrxjs-target_blankrxjsa)].
 
+## tap() vs subscribe()
+When to use tap() operator in Rx flow? Per official documentation [tap()](https://rxjs.dev/api/operators/tap) operator should be used when performing "side-effects". It means the operator should be used when user wants to "affect outside state with a notification without altering the notification". Altough side-effects can be performed inside of the map operator we should strive to write pure functions wherever possible, this is one of the cases where tap comes in handy - it can be used to make other functions pure.
+Since tap() operator can receive same callbacks as subscribe nothing stops us from transfering logic from subscribe into the corresponding (next, complete, error) callbacks, well nothing besides semantics. When writing Rx flows it is good idea to keep in mind that we wan't to convey as much information as we can just through our code, which means - use tap() for performing side effects, and debugging/logging mostly, whereas subscribe represents end of the flow which means that we should have received data which can then be used in application, this makes subscribe correct place to do something with the data you've received through piped operators. This is also why empty subcribe is basically a code smell, of course there are exceptions when you cannot avoid using empty subcsribe() e.g. triggering POST api calls via http client. 
```

Comment:
```suggestion
Since the `tap` operator's callback is the same as the subscriber's `next` callback, nothing stops us from writing the same logic in either of those places. Well, nothing besides semantics. When writing Rx flows, we should convey as much information as we can just through our code, which means - use `tap` for performing side-effects and debugging/logging mostly, whereas subscribe represents the end of the flow which means that we have probably received some data which can then be used in application. This makes subscribe callback the correct place to do something with the data you've received through piped operators. This is also why an empty subscribe call is basically a code smell. Of course, there are exceptions when you cannot avoid using an empty subscribe(), but you should be mindful of it.
```

I would remove this example from the exception:
>there are exceptions when you cannot avoid using empty sbuscribe() e.g. triggering POST API calls via the HTTP client.
Because, arguably, you still might want do to something with response data from post in subscribe callback, e.g. trigger some action like router.navigate.